tests/policies/test_interactive.py

```diff
@@ -10,44 +11,107 @@
-    interactive_policy = NoRenderingDiscreteInteractivePolicy(
-        env.observation_space,
-        env.action_space,
-        action_names,
-        action_keys,
-    )
+    interactive_policy = _get_interactive_policy(env)
+    action_keys = list(interactive_policy.action_keys_names.keys())
 
     obs = env.reset()
     done = np.array([False])
 
-    def mock_input_valid(_):
-        return random.choice(action_keys)
+    class mock_input:
+        def __init__(self):
+            self.index = 0
+
+        def __call__(self, _):
+            # Sometimes insert incorrect keys, which should get ignored by the policy.
+            if np.random.uniform() < 0.5:
+                return "invalid"
+            key = action_keys[self.index]
+            self.index = (self.index + 1) % len(action_keys)
+            return key
 
-    with patch("builtins.input", mock_input_valid):
+    with mock.patch("builtins.input", mock_input()):
+        requested_action = 0
         while not done.all():
             action, _ = interactive_policy.predict(obs)
             assert isinstance(action, np.ndarray)
             assert all(env.action_space.contains(a) for a in action)
+            assert action[0] == requested_action
 
             obs, reward, done, info = env.step(action)
             assert isinstance(obs, np.ndarray)
             assert all(env.observation_space.contains(o) for o in obs)
             assert isinstance(reward, np.ndarray)
             assert isinstance(done, np.ndarray)
+
+            requested_action = (requested_action + 1) % len(action_keys)
+
+
+@pytest.mark.parametrize("env_name", ENVS)
+def test_interactive_policy_input_validity(capsys, env_name: str):
+    """Test if appropriate feedback is given on the validity of the input."""
+    env = vec_env.DummyVecEnv([lambda: gym.wrappers.TimeLimit(gym.make(env_name), 10)])
+    env.seed(0)
+
+    interactive_policy = _get_interactive_policy(env)
+    action_keys = list(interactive_policy.action_keys_names.keys())
+
+    # Valid input key case
+    obs = env.reset()
+
+    def mock_input_valid(prompt):
+        print(prompt)
+        return action_keys[0]
+
+    with mock.patch("builtins.input", mock_input_valid):
+        interactive_policy.predict(obs)
+        stdout = capsys.readouterr().out
+        assert "Your choice" in stdout and "Invalid" not in stdout
+
+    # First invalid input key, then valid
+    obs = env.reset()
+
+    class mock_input_invalid_then_valid:
+        def __init__(self):
+            self.return_valid = False
+
+        def __call__(self, prompt):
+            print(prompt)
+            if self.return_valid:
+                return action_keys[0]
+            self.return_valid = True
+            return "invalid"
+
+    with mock.patch("builtins.input", mock_input_invalid_then_valid()):
+        interactive_policy.predict(obs)
+        stdout = capsys.readouterr().out
+        assert "Your choice" in stdout and "Invalid" in stdout
```

Comment:
Tests for `DiscreteInteractivePolicy` looks great. We're not testing `AtariInteractivePolicy` at all though. It's pretty simple granted but might still be worth testing, even if just a simple smoke test (it runs, if we feed in a key corresponding to "FIRE" we get the correct action back, etc).